### PR TITLE
Cancels

### PR DESCRIPTION
**Pull request by robbason**

### CODE REVIEW

File: app/referee/tests/test_tasks.py [hunk 1]
```diff
@@ -10,6 +10,12 @@
 from referee import scoring, tasks
 
 
+class SynchronousClient:
```
Comment (robbason) [inline app/referee/tests/test_tasks.py:13]: to me: delete this

**Review by mikemhenry — CHANGES_REQUESTED** [review]
Looks good, just have one todo

- [x] upload new ever_given version to pypi

**Comment by mikemhenry** [comment]
Sorry just saw that it is on pypi (had outdated cache when I checked)

**Review by mikemhenry — APPROVED** [review]

**Review by megosato — APPROVED** [review]

File: app/core/models.py [hunk 2]
```diff
@@ -29,11 +29,12 @@ class Status(models.TextChoices):
     PENDING = "PENDING"
     RUNNING = "RUNNING"
     CANCELLED = "CANCELLED"
+    CANCEL_PENDING = "CANCEL_PENDING"
```
[the remark this answers is not in the document]
Comment (robbason) [reply app/core/models.py:32]: at least we're not using single-letter codes, like at my other client. started with about six that made sense, C for cancel and so on... now we're almost out of letters.

File: app/core/models.py [hunk 3]
```diff
@@ -330,6 +349,24 @@ class SubmissionRun(Logged):
     def __str__(self):
         return f"{self.submission}:{self.digest}, status {self.status}"
 
+    def check_cancel_requested(self):
+        status = SubmissionRun.objects.filter(pk=self.id).values_list(
+            "status", flat=True
+        )[0]
```
[the remark this answers is not in the document]
Comment (robbason) [reply app/core/models.py:355]: figured out a better way to do this

File: app/core/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikemhenry) [reply app/core/models.py:32]: Just start using fun letters like æ :)

**Pull request closed without merging**